web/packages/shared/components/Validation/rules.ts

```diff
@@ -137,23 +189,98 @@ const validAwsIAMRoleName = (name: string): ValidationResult => {
   };
 };
 
+/**
+ * @param name validAwsRaResourceName verifies if the given value is a
+ * valid AWS Roles Anywhere resource name.
+ */
+const validAwsRaResourceName = (name: string): ValidationResult => {
+  if (name.length > ROLES_ANYWHERE_NAME_MAX_LENGTH) {
+    return {
+      valid: false,
+      message:
+        'Name should be <= ' + ROLES_ANYWHERE_NAME_MAX_LENGTH + ' characters',
+    };
+  }
+
+  if (!isRaResourceNameValid(name)) {
```

Comment:
```suggestion
  if (!name.match(ROLES_ANYWHERE_NAME_REGEX)) {
```

This check is simple enough to be inlined and the function names are very similar